xdsl/transforms/scf_for_loop_range_folding.py

```diff
@@ -62,8 +62,11 @@ def match_and_rewrite(self, op: scf.For, rewriter: PatternRewriter) -> None:
                             new_step := arith.Muli(op.step, folding_const),
                         ]
                     )
+                    new_step.result.name_hint = "new_step"
                     op.operands[2] = new_step.result
 
+            new_lb.result.name_hint = "new_lb"
+            new_ub.result.name_hint = "new_ub"
```

Comment:
This is great, but I have a feeling that it might actually make it easier to read if we keep the name hint as the user's name hint, to make it explicit what calculation we're pulling out